pyhealth/datasets/base_dataset.py

```diff
@@ -386,17 +422,22 @@ def get_all_tokens(
             raise ValueError("Please set task first.")
         tokens = []
         for sample in self.samples:
+            if type(sample[key]) == list:
+                if len(sample[key]) == 0:
+                    continue
+                # a list of lists of values
+                elif type(sample[key][0]) == list:
+                    tokens.extend(sum(sample[key], []))
+                # a list of values
+                else:
+                    tokens.extend(sample[key])
             # single value
-            if type(sample[key]) in [bool, int, str]:
-                tokens.append(sample[key])
-            # a list of values
-            elif type(sample[key][0]) in [bool, int, str]:
-                tokens.extend(sample[key])
-            # a list of lists of values
-            elif type(sample[key][0]) == list:
-                tokens.extend(sum(sample[key], []))
             else:
-                raise ValueError(f"Unknown type of {key}: {type(sample[key])}")
+                tokens.append(sample[key])
+        types = set([type(t) for t in tokens])
+        assert len(types) == 1, f"{key} tokens have mixed types"
+        assert types.pop() in [int, float, str, bool], \
```

Comment:
right now, I think we only use the `str` value?

Reply:
Yes. I try to leave some flexibility for the future (e.g., regression task)